File: 19-graphs1/1-intro/main.cpp
// Graph Introduction - C++ Example

#include <iostream>
#include <vector>
#include <unordered_map>

int main() {
    std::unordered_map<char, std::vector<char>> graph;
    graph['A'] = {'B', 'C'};
    graph['B'] = {'A', 'D'};
    graph['C'] = {'A', 'D'};
    graph['D'] = {'B', 'C'};

    std::cout << "Neighbors of node A: ";
    for (char neighbor : graph['A'])
        std::cout << neighbor << " ";
    std::cout << std::endl;
    return 0;
}

File: 19-graphs1/1-intro/main.java
// Graph Introduction - Java Example

import java.util.*;

public class GraphIntro {
    public static void main(String[] args) {
        Map<String, List<String>> graph = new HashMap<>();
        graph.put("A", Arrays.asList("B", "C"));
        graph.put("B", Arrays.asList("A", "D"));
        graph.put("C", Arrays.asList("A", "D"));
        graph.put("D", Arrays.asList("B", "C"));

        System.out.println("Neighbors of node A: " + graph.get("A"));
    }
}
